lib/models/customer_model.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+import 'package:badiup/models/user_model.dart';
+import 'package:badiup/models/user_setting_model.dart';
+import 'package:badiup/models/address_model.dart';
+
+class Customer extends User {
+  List<Address> shippingAddresses;
+
+  Customer({
+    String email,
+    String name,
+    RoleType role,
+    UserSetting setting,
+    DateTime created,
+    this.shippingAddresses,
+  }) : super(
+    email: email,
+    name: name,
+    role: role,
+    setting: setting,
+    created: created,
+  );
+
+  @override
+  Map<String, dynamic> toMap() {
+    Map<String, dynamic> customerMap = super.toMap();
+    customerMap['shippingAddresses'] = shippingAddresses.map(
```

Comment:
Since we're storing the shippingAddresses, I don't think we need this piece of code.